src/workerd/api/html-rewriter.h

```diff
@@ -304,6 +307,8 @@ class EndTag final: public HTMLRewriter::Token {
 
  private:
   kj::Maybe<CType&> impl;
+  // TODO(npaun): is adding rewriter field like this legit?
```

Comment:
I would say, not quite legit, because this could become dangling. The reason is that subclasses of Token are the C++-side implementations of JS wrapper objects, meaning JS can arbitrarily extend their lifetimes beyond `Rewriter`'s lifetime. This is the reason there's the TokenScope guard which nullifies `impl` above.

Of course, `checkToken()` is called in all of the places we use this `rewriter` reference, so the code as written _is_ safe, but it feels like leaving a footgun around.

It occurs to me that an alternative could be to make each Token subclass own their own HashMap of registered replacers. We would expect each Token's replacers to be dropped before its associated TokenScope ends, meaning we could have `~TokenScope()` assert if there are any registered replacers left in the Token.

Reply:
Thanks 🙏🏻👍🏻

Reply:
The dangling reference issue is fixed.
However, I didn't move the registered replacers to `Token` yet because it seems like the replacer callback can be invoked after their registering handler has exited. Happy to revisit this if you have any ideas to try later on.